src/utils/issue.ts

```diff
@@ -0,0 +1,200 @@
+    issueComments = await fetchIssueComments(params);
+  }
+
+  const {
+    context: {
+      logger,
+      payload: {
+        repository: {
+          owner: { login },
+        },
+      },
+    },
+  } = params;
+
+  if (!issueComments) {
+    logger.info("No comments found on issue");
+    return linkedIssues;
+  }
+
+  for (const comment of issueComments) {
+    const linkedIssue = idIssueFromComment(login, comment.body);
+    if (linkedIssue) {
+      linkedIssues.push(linkedIssue);
+    }
+  }
+
+  return await filterLinkedIssues(params, linkedIssues);
+}
+
+async function filterLinkedIssues(params: FetchParams, linkedIssues: { issueNumber: number; repo: string }[]) {
+  const {
+    context: {
+      logger,
+      payload: {
+        repository: {
+          owner: { login },
+        },
+      },
+    },
+  } = params;
+
+  const contextIssues: {
+    issueNumber: number;
+    repo: string;
+  }[] = [];
+
+  for (const issue of linkedIssues) {
+    if (issue && issue.issueNumber && issue.repo) {
+      if (await isRepoFromSameOrg(params.context, issue.repo, login)) {
+        contextIssues.push({
+          issueNumber: issue.issueNumber,
+          repo: issue.repo,
+        });
+      } else {
+        logger.info(`Ignoring linked issue ${issue.issueNumber} from ${issue.repo} as it is not from the same org`);
+      }
+    }
+  }
+
+  return contextIssues;
+}
+
+export async function getLinkedIssueContextFromComments(context: Context, issueComments: IssueComments) {
+  // find any linked issues in comments by parsing the comments and enforcing that the
+  // linked issue is from the same org that the current issue is from
+  const linkedIssues = await fetchLinkedIssues({ context }, issueComments);
+
+  // the conversational history of the linked issues
+  const linkedIssueComments: IssueComments = [];
+
+  // we are only going one level deep with the linked issue context fetching
+  for (const issue of linkedIssues) {
+    const fetched = await fetchIssueComments({ context, issueNum: issue.issueNumber, repo: issue.repo });
+    linkedIssueComments.push(...fetched);
+  }
+
+  return { linkedIssues, linkedIssueComments };
+}
+
+export function idIssueFromComment(owner?: string, comment?: string | null) {
+  if (!comment) {
+    return null;
+  }
+  if (!owner) {
+    throw new Error("Owner must be provided when parsing linked issues");
+  }
+  // the assumption here is that any special GitHub markdown formatting is converted to an anchor tag
+  const urlMatch = comment.match(/https:\/\/github.com\/([^/]+)\/([^/]+)\/(pull|issue|issues)\/(\d+)/);
+
+  /**
+   * I think we should restrict including any linked context which is not of the same org.
+   *
+   * In most cases this will be the expected behaviour, I remember a scenario where
+   * I linked to an issue in a 3rd party org, for extra reviewer context but I also include the
+   * TL;DR which is always the case. We wouldn't want that full 3rd party PR review or issue to be
```

Comment:
Just be sure to label the section with the org/repo name and ChatGPT should know it's not as relevant